### DISCUSSION TITLE
Existing wallet login w/ seed phrase

### DISCUSSION THREAD

**Discussion by ryunik** [Web3Auth/Web3Auth#551] [Feature Requests]
### **Is your feature request related to a problem? Please describe.**

A way for users with existing wallets (eg. metamask) to use the same flow to connect their wallet via seed phrase

### **Describe the solution you'd like**

Adapter which allows users to input a seed phrase, which is stored encrypted locally on their computer. This does not have to convert their PK to be usable with openlogin, just an option to allow existing crypto users to not have to create a new wallet for apps that use the web3 auth flow. 

### **Describe alternatives you've considered**

Separate integration to have the user input an existing seed phrase. This for sure works, but would be good to have all users in the same single auth flow. I'm guessing this could be accomplished by writing your own adapter, but I'd imagine that this would be a pretty common task for all apps looking to use something like web3auth. Currently AFAIK all existing login methods for web3auth require you to make a new account rather than use an existing wallet. There are External wallet options for Metamask or Coinbase extension, but the current use case I'm exploring is for use in an Electron where neither of the 2 are available.

### **Additional context**

The web3auth site seems to indicate that this is possible, but I might have just missed how this is done:

<img width="843" alt="image" src="https://user-images.githubusercontent.com/106510479/173695332-eeead087-b298-4887-8cc1-5631da11dbb8.png">
https://web3auth.io/

<img width="879" alt="image" src="https://user-images.githubusercontent.com/106510479/173695654-b4476dfa-7889-4406-afde-bff5e4a0eeab.png">
https://docs.web3auth.io/developing-with-web3auth/adapters#currently-available-adapters

**Comment by yashovardhan** [comment]
Hey Nick,

Sorry for not getting back to you promptly around this question. We are constantly developing new adapters. Somehow this particular adapter was not prioritized according to our community needs. We'll add this to our development cycle as soon as possible and update you accordingly.


In the meantime you can utilise the following options of implementing it directly:
- Web3Auth never saves your private key within any server, it is dynamically reconstructed within your application. You can use the [Web3Auth Self Host](https://web3auth.io/docs/guides/selfhost) to get your user to input a private key and save the shares somewhere within your servers and dynamically reconstruct it within your frontend. You can also use the browser local storage to share keys (as Web3Auth does with the device share) and utlise it directly. There are dedicated modules available for that.
- For this login, we can expect a user to enter their private key on every new session, you can safely store their private key within the session storage.
- For directly using the private key for making transactions, you can refer to the particular documentation of the chain you're building upon. You can also refer to a few common chains that we have highlighted in our [documentation](https://web3auth.io/docs/connect-blockchain/)
- For EVM and solana chains, web3auth has a dedicated provider. This provider can be imported as separate modules as [`ethereum-provider`](https://web3auth.io/docs/sdk/web/providers/evm#getting-a-provider-from-any-secp512k1-private-key) and [`solana-provider`](https://web3auth.io/docs/sdk/web/providers/solana#getting-a-provider-from-any-secp512k1-private-key) where you can pass your private key and get a provider for making RPC Calls.
